sympy/core/tests/test_function.py

```diff
@@ -23,6 +23,25 @@ def test_f_expand_complex():
     assert exp(z).expand(complex=True) == cos(im(z))*exp(re(z)) + \
         I*sin(im(z))*exp(re(z))
 
+def test_unicode_name_support():
+    Function(u('k')) # check if unicoded function names can be created
+
+    # check if unicoded function is same as its ascii counterpart
+    assert Function('k') == Function(u('k'))
+
+def test_greek_name_support():
+    Function('α')
+    Function(u('α'))
+    Function('α') == Function(u('α'))
+
+    Function('αβ')
+    Function(u('αβ'))
+    Function('αβ') == Function(u('αβ'))
+
+    Function('aαβ')
+    Function(u('aαβ'))
+    Function('aαβ') == Function(u('aαβ'))
+
```

Comment:
The purpose of the first three tests I understand (though it's a bit dubious to label them as "test Greek", "non-ASCII characters" is actually required).
The purpose of the six tests after that I do not understand. What kinds of potential errors do they catch? Can this be documented?